bin/yarn.js

```diff
@@ -4,41 +4,31 @@
 /* eslint-disable flowtype/require-valid-file-annotation */
 'use strict';
 
-// validate that used node version is supported
-var semver = require('semver');
 var ver = process.versions.node;
-ver = ver.split('-')[0]; // explode and truncate tag from version #511
+var majorVer = parseInt(ver.split('.')[0], 10);
 
 var dirPath = null;
 
-if (semver.satisfies(ver, '>=5.0.0')) {
-  dirPath = '../lib';
-} else if (semver.satisfies(ver, '>=4.0.0')) {
-  dirPath = '../lib-legacy';
+if (majorVer >= 5) {
+  dirPath = '../lib/';
+} else if (majorVer >= 4) {
+  dirPath = '../lib-legacy/';
 } else {
-  console.log(require('chalk').red('Node version ' + ver + ' is not supported, please use Node.js 4.0 or higher.'));
-  // We need to skip the rest and terminate here since the Node version is not supported.
-  // `console.log` calls are sync so we don't have a risk of exiting early before stdout flushes
-  process.exit(1); // eslint-disable-line no-process-exit
+  console.error('Node version ' + ver + ' is not supported, please use Node.js 4.0 or higher.');
+  process.exitCode = 1;
```

Comment:
This seems like a legit place to call `process.exit`

Don't you need a `return` here? Otherwise it'll keep running even after hitting an incompatible Node.js version.